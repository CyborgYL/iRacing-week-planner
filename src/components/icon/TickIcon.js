import React from 'react';
import styles from '../../styles/main.module.scss';

export default function TickIcon() {
  return <span className={`${styles.glyphicon} ${styles['glyphicon-ok']}`} />;
}
