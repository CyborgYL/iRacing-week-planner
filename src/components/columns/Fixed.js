// @flow

import * as React from 'react';
import TickIcon from '../icon/TickIcon';

type Props = {
  race: {
    fixed: boolean,
  }
};

export default function Fixed({ race }: Props) {
  return (
    <td>
      <div>
        {race.fixed && <TickIcon />}
      </div>
    </td>
  );
}
