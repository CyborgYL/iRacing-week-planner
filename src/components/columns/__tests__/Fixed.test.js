import React from 'react';
import { shallow } from 'enzyme';
import { describe, test } from '@jest/globals';
import TickIcon from '../../icon/TickIcon';

import Fixed from '../Fixed';

describe('components/columns/Fixed', () => {
  test('renders correctly', () => {
    const component = shallow(<Fixed race={{ fixed: true }} />);

    expect(component).toMatchSnapshot();
    expect(component.find(TickIcon).length).toBe(1);

    const component2 = shallow(<Fixed race={{ fixed: false }} />);

    expect(component2).toMatchSnapshot();
    expect(component2.find(TickIcon).length).toBe(0);
  });
});
