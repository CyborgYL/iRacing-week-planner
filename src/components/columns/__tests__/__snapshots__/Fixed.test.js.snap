// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/columns/Fixed renders correctly 1`] = `
<td>
  <div>
    <TickIcon />
  </div>
</td>
`;

exports[`components/columns/Fixed renders correctly 2`] = `
<td>
  <div />
</td>
`;
